### PR TITLE
SQL: added encoding file

### PR DESCRIPTION
**Pull request by schnek**
updated list FlightMaster
adding macro DBC_TAXI_MASK_SIZE
fixed build for MOP
AttackerLvl up from MOP.

### CODE REVIEW

File: sql/world/updates/20180817-00_guild_data.sql [hunk 1]
```diff
@@ -12,7 +12,7 @@ CREATE TABLE IF NOT EXISTS `guild_rewards` (
   `price` bigint(20) unsigned DEFAULT '0',
   `achievement` int(10) unsigned DEFAULT '0',
   PRIMARY KEY (`entry`)
-) ENGINE=MyISAM DEFAULT CHARSET=utf8;
+) ENGINE=MyISAM DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
```
Comment (Zyres) [inline sql/world/updates/20180817-00_guild_data.sql]: We can not change old update files.

File: sql/world/updates/20180817-00_guild_data.sql [hunk 2]
```diff
@@ -69,7 +69,7 @@ CREATE TABLE IF NOT EXISTS `guild_xp_for_level` (
   `lvl` int(11) unsigned NOT NULL,
   `xp_for_next_level` bigint(20) unsigned NOT NULL DEFAULT '0',
   PRIMARY KEY (`lvl`)
-) ENGINE=MyISAM DEFAULT CHARSET=utf8;
+) ENGINE=MyISAM DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
```
Comment (Zyres) [inline sql/world/updates/20180817-00_guild_data.sql]: We can not change old update files.

File: sql/world/updates/20181208-00_spell_coefficient_override.sql [hunk 3]
```diff
@@ -9,7 +9,7 @@ CREATE TABLE IF NOT EXISTS `spell_coefficient_override` (
   PRIMARY KEY (`spell_id`,`min_build`),
   UNIQUE KEY `unique_index` (`spell_id`,`min_build`) USING BTREE,
   KEY `spell_id` (`spell_id`)
-) ENGINE=InnoDB DEFAULT CHARSET=utf8 COMMENT='Spell System';
+) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci; COMMENT='Spell System';
```
Comment (Zyres) [inline sql/world/updates/20181208-00_spell_coefficient_override.sql]: We can not change old update files.

File: src/world/GameMop/Management/CMakeLists.txt [hunk 4]
```diff
@@ -5,7 +5,7 @@ set(PATH_PREFIX GameMop/Management)
 set(SRC_GAMEMOP_MANAGEMENT_FILES
     ${PATH_PREFIX}/GuildFinderMgr.cpp
     ${PATH_PREFIX}/GuildFinderMgr.h
-    ${PATH_PREFIX}/QuestMgr.Cata.cpp
+    # ${PATH_PREFIX}/QuestMgr.Cata.cpp
```
Comment (Zyres) [inline src/world/GameMop/Management/CMakeLists.txt]: If this file is no longer used, delete this line.

File: src/world/Units/Players/Player.h [hunk 5]
```diff
@@ -1072,15 +1072,15 @@ class SERVER_DECL Player : public Unit
         /////////////////////////////////////////////////////////////////////////////////////////
         TaxiPath* GetTaxiPath() { return m_CurrentTaxiPath; }
         bool isOnTaxi() const { return m_onTaxi; }
-        const uint32 & GetTaximask(uint8 index) const { return m_taximask[index]; }
+        const uint32 & GetTaximask(uint32 index) const { return m_taximask[index]; }
```
Comment (Zyres) [inline src/world/Units/Players/Player.h]: use fixed size types (uint32_t)

File: src/world/Units/Players/Player.h [hunk 6]
```diff
@@ -1072,15 +1072,15 @@ class SERVER_DECL Player : public Unit
...
         void LoadTaxiMask(const char* data);
         void TaxiStart(TaxiPath* path, uint32 modelid, uint32 start_node);
         void JumpToEndTaxiNode(TaxiPath* path);
         void EventDismount(uint32 money, float x, float y, float z);
         void EventTaxiInterpolate();
 
         void SetTaxiState(bool state) { m_onTaxi = state; }
-        void SetTaximask(uint8 index, uint32 value) { m_taximask[index] = value; }
+        void SetTaximask(uint32 index, uint32 value) { m_taximask[index] = value; }
```
Comment (Zyres) [inline src/world/Units/Players/Player.h]: use fixed size types (uint32_t)

File: sql/world/updates/20181208-00_spell_coefficient_override.sql (hunk 3 above)
Comment (schnek) [reply sql/world/updates/20181208-00_spell_coefficient_override.sql]: `Sql query failed due to [You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ', assign_on_target_flag, assign_self_cast_only, assign_c_is_flag, proc_flags, pr' at line 1], Query: [SELECT spell_id, rank, assign_on_target_flag, assign_self_cast_only, assign_c_is_flag, proc_flags, proc_target_selfs, proc_chance, proc_charges, proc_interval, proc_effect_trigger_spell_0, proc_effect_trigger_spell_1, proc_effect_trigger_spell_2 FROM spell_custom_override]`

some update does not load

Comment (schnek) [reply sql/world/updates/20181208-00_spell_coefficient_override.sql]: removed mention from rt

File: src/world/GameMop/Management/CMakeLists.txt (hunk 4 above)
Comment (schnek) [reply src/world/GameMop/Management/CMakeLists.txt]: delete file (QuestMgr.Cata.cpp)

File: src/world/Units/Players/Player.h (hunk 5 above)
Comment (schnek) [reply src/world/Units/Players/Player.h]: up

File: src/world/Units/Players/Player.h (hunk 6 above)
Comment (schnek) [reply src/world/Units/Players/Player.h]: up

**Comment by Zyres** [comment]
Hey schnek, is utf8mb4_unicode_ci even needed? What about utf8_general_ci?
Why do we need the collation in account_permissions ...

Does it even work for ruRU with your changes or do we still miss something to send these strings in the correct format?

**Comment by schnek** [comment]
> Hey schnek, is utf8mb4_unicode_ci even needed? What about utf8_general_ci?

As far as Latin (ie "European") languages go, there is not much difference between the Unicode sorting and the simplified utf8mb4_general_ci sorting in MySQL, but there are still a few differences:

For examples, the Unicode collation sorts "ß" like "ss", and "Œ" like "OE" as people using those characters would normally want, whereas utf8mb4_general_ci sorts them as single characters (presumably like "s" and "e" respectively).

Some Unicode characters are defined as ignorable, which means they shouldn't count toward the sort order and the comparison should move on to the next character instead. utf8mb4_unicode_ci handles these properly. [//stackoverflow](https://stackoverflow.com/questions/766809/whats-the-difference-between-utf8-general-ci-and-utf8-unicode-ci)

> Does it even work for ruRU with your changes or do we still miss something to send these strings in the correct format?

for ruRU it works as it should.

> Why do we need the collation in account_permissions ...

maybe it's useless. the console does not support Russian.
```
Account create русский русский
usage: account create <name> <password> <email>
example: account create ghostcrawler Ih4t3p4l4dins greg.street@blizzard.com
```

**Pull request merged**